src/main/kotlin/app/revanced/patches/youtube/misc/settings/framework/components/impl/ListPreference.kt

```diff
@@ -0,0 +1,11 @@
+package app.revanced.patches.youtube.misc.settings.framework.components.impl
+
+import app.revanced.patches.youtube.misc.settings.framework.components.BasePreference
+
+internal class ListPreference(

```

Comment:
A list preference should contain a property for `elements`. That way you can define a list preference with elements. For this PR you would then create a `ListPreference` for the video qualities.

Reply:
What do you mean by `elements`? Please elaborate!
Explain your suggestion thoroughly, as I am a beginner in field.

Reply:
```diff
internal class ListPreference(
    key : String,
    title : StringResource,
+   val elements: ArrayResource,
    val summary : StringResource
): BasePreference(key, title) {
    override val tag: String = "ListPreference"
}
```

This way you can then create:
```kt

ListPreference(elements = ArrayResource("name", listOf("144p", "1080p")))

Reply:
**It's not working:**
elements = ArrayResource(
                            "revanced_video_quality_list_wifi",
                            listOf(
                                StringResource("Auto","-2"),
                                StringResource("144p","144"),
                                StringResource("240p","240"),
                                StringResource("360p","360"),
                                StringResource("480p","480"),
                                StringResource("720p","720"),
                                StringResource("1080p","1080"),
                                StringResource("1440p","1440"),
                                StringResource("2160p","2160")
                            )
                        ),

Reply:
Its a while since I implemented `ArrayResource`, probably it serves a different purpose.

Reply:
Should I stick with my current implementation?

Reply:
Refer to https://stackoverflow.com/questions/9880841/using-list-preference-in-android. it shows which abstractions are rewuired to create the list of qualities for the ListPreference. Seems like you can define a list of elements by referencing it in the ListPreference properties.

Reply:
ListPreference is now available with 0533e6c63e8da02f0b2b4df9652450c178255215.